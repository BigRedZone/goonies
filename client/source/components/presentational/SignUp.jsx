import React from 'react';
import axios from 'axios';
import { Redirect } from '@reach/router';


class SignUp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      experience: 'Novice',
      loggedIn: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  handleSubmit() {
    const { username, password, experience } = this.state;
    const userInfo = { username, password, experience };
    axios.post('/api/users/create', userInfo)
      .then(function (response) {
        if (response.data === 'Username Taken') {
          alert('That username is already taken. Please choose another username.')
        } else {
          //  success! redirect
          this.setState({
            loggedIn: true,
          });
        }
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    //  destructuring state object per airbnb syntax guide
    const { loggedIn, experience } = this.state;
    //  if user successfully signs up, redirect to maps page
    if (loggedIn) {
      return <Redirect noThrow to="/maps" />;
    }
    return (
      <div>
        <form>
          Username:
          <input type="text" name="username" onChange={this.handleChange} />
          <br />
          Password:
          <input type="password" name="password" onChange={this.handleChange} />
          <br />
            Experience Level:
          <select name="experience" value={experience} onChange={this.handleChange}>
            <option value="Novice">Novice</option>
            <option value="Intermediate">Intermediate</option>
            <option value="Advanced">Advanced</option>
          </select>
          <br />
          <input type="button" value="Submit" onClick={this.handleSubmit} />
        </form>
      </div>
    );
  }
}

export default SignUp;
