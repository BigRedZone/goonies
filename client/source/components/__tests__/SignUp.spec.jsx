import React from 'react';
import renderer from 'react-test-renderer';
import SignUp from '../presentational/SignUp';

test('App Renders correctly', () => {
  const component = renderer.create(<SignUp />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
