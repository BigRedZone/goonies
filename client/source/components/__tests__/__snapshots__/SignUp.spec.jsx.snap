// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Renders correctly 1`] = `
<div
  className="login"
>
  <form>
    <div
      className="loginfield"
    >
      <label
        htmlFor="username"
      >
        Username:
        <br />
        <input
          name="username"
          onChange={[Function]}
          type="text"
        />
      </label>
      <br />
      <label
        htmlFor="password"
      >
        Password:
        <br />
        <input
          name="password"
          onChange={[Function]}
          type="password"
        />
      </label>
      <br />
      <p>
        Experience Level:
      </p>
      <select
        name="experience"
        onChange={[Function]}
        value="Novice"
      >
        <option
          value="Novice"
        >
          Novice
        </option>
        <option
          value="Intermediate"
        >
          Intermediate
        </option>
        <option
          value="Advanced"
        >
          Advanced
        </option>
      </select>
      <br />
      <input
        className="signup"
        onClick={[Function]}
        type="button"
        value="Submit"
      />
    </div>
  </form>
</div>
`;
